src/renderer/pages/Operations/components/modals/ApproveTx.tsx

```diff
@@ -237,29 +235,36 @@ const ApproveTx = ({ tx, account, connection }: Props) => {
   }
 
   const isSubmitStep = activeStep === Step.SUBMIT && approveTx && signAccount && signature && txPayload;
+  if (isSubmitStep && connection.api) {
+    return (
+      <Submit
+        tx={approveTx}
+        api={connection.api}
+        multisigTx={tx}
+        account={signAccount}
+        txPayload={txPayload}
+        signature={signature}
+        onClose={handleClose}
+      />
+    );
+  }
 
   return (
-    <>
-      <Button className="ml-auto" onClick={() => setIsModalOpen(true)}>
-        {t('operation.approveButton')}
-      </Button>
-
-      <BaseModal
-        closeButton
-        isOpen={activeStep !== Step.SUBMIT && isModalOpen}
-        title={<OperationTitle title={t(transactionTitle, { asset: asset?.symbol })} chainId={tx.chainId} />}
-        contentClass={activeStep === Step.SIGNING ? '' : undefined}
-        headerClass="py-3 pl-5 pr-3"
-        panelClass="w-[440px]"
-        onClose={handleClose}
-      >
+    <Modal size="md" onToggle={handleClose}>
+      <Modal.Trigger>
+        <Button className="ml-auto">{t('operation.approveButton')}</Button>
```

Comment:
You can pass button as children inside ApproveTxModal component, it should be clearer for reading